server/testdata/tools/xlam.gotmpl

```diff
@@ -0,0 +1,47 @@
+{{ if .System }}{{ .System }}
+{{ end }}
+{{- range $i, $m := .Messages }}
+{{- if eq .Role "user" }}
+{{- $last := (or (eq (len (slice $.Messages $i)) 1) (eq (len (slice $.Messages $i)) 2)) -}}
+### Instruction:
+{{- if and $.Tools $last }}
```

Comment:
`$last` can be replaced with `(le (len (slice $.Messages $i)) 2)`

Reply:
amazing, thanks